src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +227,26 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Calculates the left curly corresponding to the block to be checked.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return the left curly corresponding to the block to be checked
+     */
+    private static DetailAST findLeftCurly(DetailAST ast) {
+        DetailAST leftCurly;
+        if (ast.findFirstToken(TokenTypes.SLIST) == null) {
+            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        }
+        else {
+            leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+        }
+        if (ast.getType() == TokenTypes.LITERAL_CASE
+                && ast.getNextSibling() != null
+                && "{".equals(ast.getNextSibling().getFirstChild().getText())) {
```

Comment:
@romani 
Exactly ! Also, I said 
>case token would have its next "sibling" as a slist ast... slist ast would have it's first child of type slist ast. 

I acknowledge `LITERAL_CASE` never has SLIST under it.
That's what I have been trying to say that to inspect `case` token we are having to investigate its sibling and not children ! I was also feeling that there's something wrong in the way it is coded. CASE_GROUP should be used. But STILL, like I pointed out if you do `ast.findFirstToken(TokenTypes.SLIST)` even for a CASE_GROUP, you are bound to get `NullPointerException` in `hasText()` one way or the other. 

@MEZk 
Yes, `hasText` seems to have issues because it was giving `NullPointerException` even for `SLIST` for which it was designed. But `findRcurly` seems like extra work which can be deferred until some situation makes it absolutely necessary to have `findRightCurly` . Only `case` seems to be the only exception as of now which can be handled with just `findLeftCurly`. 

So what do you guys suggest ?